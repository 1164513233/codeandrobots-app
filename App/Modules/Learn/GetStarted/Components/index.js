import React, { Component } from 'react'

import {
  Container,
  Carousel,
  Card } from 'App/Components'

import { Images } from 'App/Themes'

export default class Screen extends Component {
  onDone = () => {
    this.props.navigation.goBack()
  }

  onLearnMorePress = () => {
    this.props.navigation.navigate('WebScreen', {
      source: 'http://www.codeandrobots.com'
    })
  }

  render () {
    return (
      <Container>
        <Carousel onDone={this.onDone}>
          <Card image={Images.hello} title='Hi there ðŸ‘‹' />
          <Card
            image={Images.hello}
            text={'Sorry, the app is still under construction ðŸš§\n\nThe good news is you can follow along as this Beta version of the app develops over the coming weeks.'}
            button='Learn More'
            onPress={this.onLearnMorePress} />
        </Carousel>
      </Container>
    )
  }
}
