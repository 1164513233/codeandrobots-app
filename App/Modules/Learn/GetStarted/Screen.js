import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  Container,
  Carousel,
  Card } from 'App/Components'

import { Images } from 'App/Themes'

export default class Screen extends Component {
  static propTypes = {
    onDone: PropTypes.func.isRequired,
    onLearnMorePress: PropTypes.func.isRequired
  }

  render () {
    const { onDone, onLearnMorePress } = this.props
    return (
      <Container>
        <Carousel onDone={onDone}>
          <Card image={Images.hello} title='Hi there ðŸ‘‹' />
          <Card
            image={Images.hello}
            text={'Sorry, the app is still under construction ðŸš§\n\nThe good news is you can follow along as this Beta version of the app develops over the coming weeks.'}
            button='Learn More'
            onPress={onLearnMorePress} />
        </Carousel>
      </Container>
    )
  }
}
