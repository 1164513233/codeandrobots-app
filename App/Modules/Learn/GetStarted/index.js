import React, { Component } from 'react'
import { connect } from 'react-redux'

import Screen from './Screen'

export class GetStartedContainer extends Component {
  onDone = () => {
    this.props.navigation.goBack()
  }

  onLearnMorePress = () => {
    this.props.navigation.navigate('WebScreen', {
      source: 'http://www.codeandrobots.com'
    })
  }

  render () {
    return (
      <Screen
        ref={(ref) => {
          this.screen = ref
        }}
        {...this.props}
        onDone={this.onDone}
        onLearnMorePress={this.onLearnMorePress}
      />
    )
  }
}

const mapStateToProps = (state) => {
  return {
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(GetStartedContainer)
