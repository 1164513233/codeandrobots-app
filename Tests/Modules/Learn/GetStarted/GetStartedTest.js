import React from 'react'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import GetStarted from 'App/Modules/Learn/GetStarted/Components'

describe('GetStarted', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <GetStarted />)
  })

  it('Component renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
