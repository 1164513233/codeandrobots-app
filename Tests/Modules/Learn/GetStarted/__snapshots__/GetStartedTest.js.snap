// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetStarted Component renders correctly 1`] = `
<Container>
  <Carousel
    onDone={[Function]}
  >
    <Card
      image={
        Object {
          "process": [Function],
        }
      }
      title="Hi there ðŸ‘‹"
    />
    <Card
      button="Learn More"
      image={
        Object {
          "process": [Function],
        }
      }
      onPress={[Function]}
      text="Sorry, the app is still under construction ðŸš§

The good news is you can follow along as this Beta version of the app develops over the coming weeks."
    />
  </Carousel>
</Container>
`;
